src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java

```diff
@@ -936,6 +985,166 @@ void testGetApprovedDatasetsWhenEmpty() {
+    voteDAO.updateVote(true, "rationale", today, voteId4, false, electionId4, today, false);
+    electionDAO.updateElectionById(electionId4, ElectionStatus.CLOSED.getValue(), today);
+
+    List<ApprovedDataset> approvedDatasets = datasetDAO.getApprovedDatasets(user.getUserId());
+    assertNotNull(approvedDatasets);
+    assertEquals(4, approvedDatasets.size());
+
+    // manually age the dar off
+    jdbi.useHandle(handle -> {
+      handle.createUpdate("UPDATE data_access_request SET submission_date=CURRENT_DATE - INTERVAL '365 days' WHERE data_access_request.reference_id = ?").bind(0, darCollection.getMostRecentDar().getReferenceId()).execute();
+    });
+
+    // confirm the expiration part of the query works.
+    List<ApprovedDataset> approvedDatasets2 = datasetDAO.getApprovedDatasets(user.getUserId());
+    assertEquals(0, approvedDatasets2.size());
+
+    DataAccessRequest recentDar = darCollection.getMostRecentDar();
+
+    // submit a progress report.
+    DataAccessRequest progressReport = createProgressReport(recentDar.getData(), recentDar.getEraCommonsId(), recentDar.getUserId(), recentDar.getCollectionId(), recentDar.getId(), recentDar.getDatasetIds());
+
+    // ensure we still have no approved datasets.
+    List<ApprovedDataset> approvedDatasets3 = datasetDAO.getApprovedDatasets(user.getUserId());
+    assertEquals(0, approvedDatasets3.size());
+
+    // Simulate 2 DAC 2 elections for today
+    // vote yes on dataset 3
+    Integer electionId5 = electionDAO.insertElection(
+        ElectionType.DATA_ACCESS.getValue(),
+        ElectionStatus.OPEN.getValue(),
+        today,
+        progressReport.getReferenceId(),
+        dataset3.getDatasetId()
+    );
+    Integer voteId5 = voteDAO.insertVote(chairperson2.getUserId(), electionId5, VoteType.FINAL.getValue());
+    voteDAO.updateVote(true, "rationale", today, voteId5, false, electionId5, today, false);
+    electionDAO.updateElectionById(electionId5, ElectionStatus.CLOSED.getValue(), today);
+    assertEquals(1, datasetDAO.getApprovedDatasets(user.getUserId()).size());
+
+    Integer electionId6 = electionDAO.insertElection(
+        ElectionType.DATA_ACCESS.getValue(),
+        ElectionStatus.OPEN.getValue(),
+        today,
+        progressReport.getReferenceId(),
+        dataset4.getDatasetId()
+    );
+    Integer voteId6 = voteDAO.insertVote(chairperson2.getUserId(), electionId6, VoteType.FINAL.getValue());
+    // vote no on dataset 4
+    voteDAO.updateVote(false, "rationale", today, voteId6, false, electionId6, today, false);
+    electionDAO.updateElectionById(electionId6, ElectionStatus.CLOSED.getValue(), today);
+
+    List<ApprovedDataset> approvedDatasets4 = datasetDAO.getApprovedDatasets(user.getUserId());
+    assertNotNull(approvedDatasets4);
+    assertEquals(1, approvedDatasets4.size());
+
+    List<DataAccessRequest> dataset3Dars = dataAccessRequestDAO.findApprovedDARsByDatasetId(dataset3.getDatasetId());
```

Comment:
Here's where I cross concerns and start looking at the query the TDR resource leverages.